spyder_kernels/console/shell.py

```diff
@@ -45,8 +48,8 @@ class SpyderShell(ZMQInteractiveShell):
     ]
 
     def __init__(self, *args, **kwargs):
-        # Create _pdb_obj_stack before __init__
-        self._pdb_obj_stack = []
```

Comment:
Both the debugger and the namespace manager are in this stack so we know which is the current context